lib/usbd/usbd.c

```diff
@@ -224,6 +234,8 @@ bool usbd_is_vbus(usbd_device *dev){
 }
 
 void usbd_enable(usbd_device *dev, bool onoff){
+	if (!onoff)
+		usbd_disconnect(dev, false);
 	if (dev->backend->power_control)
 		dev->backend->power_control(dev, (onoff)? usbd_paActivate : usbd_paShutdown );
 }
```

Comment:
yes, this was confused me too. disconnection MUST be provided before powerdown. but fro stm DWC core it is mean that USB enter to non-connectable state. imho, name <disconnect> - not very suitable for this operation.
i change behaviour to restore disconnection state on powerdown complete. 
i not sure, is it good to place disconnection before powerdown into backend->power_control, instead of botherig it on frontend?